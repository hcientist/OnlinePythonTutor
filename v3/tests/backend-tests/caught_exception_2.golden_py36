{
  "code": "# caught exception:\ntry:\n  x = 1 / 0\nexcept:\n  print(\"DIVIDE BY ZERO\")\n\n\n# uncaught:\nprint(y)\n\nprint(\"should not reach here\")\n\n",
  "trace": [
    {
      "line": 2,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": ""
    },
    {
      "line": 3,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": ""
    },
    {
      "line": 3,
      "event": "exception",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": "",
      "exception_msg": "ZeroDivisionError: division by zero"
    },
    {
      "line": 4,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": ""
    },
    {
      "line": 5,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": ""
    },
    {
      "line": 9,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": "DIVIDE BY ZERO\n"
    },
    {
      "line": 9,
      "event": "exception",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": "DIVIDE BY ZERO\n",
      "exception_msg": "NameError: name 'y' is not defined"
    }
  ]
}
