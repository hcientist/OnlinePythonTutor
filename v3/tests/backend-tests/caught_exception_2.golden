{
  "code": "# caught exception:\ntry:\n  x = 1 / 0\nexcept:\n  print(\"DIVIDE BY ZERO\")\n\n\n# uncaught:\nprint(y)\n\nprint(\"should not reach here\")\n\n", 
  "trace": [
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 2, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 3, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [], 
      "stdout": "", 
      "exception_msg": "ZeroDivisionError: integer division or modulo by zero", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 3, 
      "event": "exception"
    }, 
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 4, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 5, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [], 
      "stdout": "DIVIDE BY ZERO\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 9, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [], 
      "stdout": "DIVIDE BY ZERO\n", 
      "exception_msg": "NameError: name 'y' is not defined", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 9, 
      "event": "exception"
    }
  ]
}
